docs/book/features/vsphere_csi_migration.md

```diff
@@ -0,0 +1,122 @@
+<!-- markdownlint-disable MD033 -->
+<!-- markdownlint-disable MD034 -->
+# vSphere CSI Driver - vSphere CSI Migration
+
+**Note:** vSphere CSI Driver images for this feature is not yet released. When images will be available for public consumption we will update this document.
+
+vSphere CSI driver and CNS bring in a lot of features that are not available in the in-tree vSphere volume plugin.
+
+Refer to the following feature comparisons table to know what is added in the vSpehre CSI driver.
+
+| Feature | In-tree vSphere Volume Plugin | vSphere CSI Driver |
+|---------|-------------------------------|--------------------|
+| Block volume (`ReadWriteOnce` access mode) | Supported. Block volumes are backed by vmdks. | Supported. Block volumes are backed by vSphere Improved Virtual Disk(management layer on top of vmdk). |
+| File volume (`ReadWriteMany`/`ReadOnlyMany` access modes) | Not Supported | Supported. File volumes are backed by vSAN file shares. |
+| Dynamic provisioning(with and without SPBM) | Supported for block volumes only | Supported for block and file volumes |
+| Static Provisioning | Supported for block volumes only | Supported for block and file volumes |
+| Expand volume | Not supported | Supported for block volumes |
+| Storage vMotion of block volumes | Not supported | Supported |
+| vSphere UI integration(CNS dashboard, vSAN virtual objects, vSphere space) | Not supported | Supported |
+| [vSphere with Kubernetes Configuration and Management](https://docs.vmware.com/en/VMware-vSphere/7.0/vmware-vsphere-with-kubernetes/GUID-152BE7D2-E227-4DAA-B527-557B564D9718.html) | Not supported | Supported |
```

Comment:
Can we call this feature as "vSphere with Kubernetes integration"?

Reply:
done.